app-emulation/wine-proton/wine-proton-7.0.4.ebuild

```diff
@@ -0,0 +1,303 @@
+		dlls/appwiz.cpl/addons.c || die)"
+	[[ ${WINE_GECKO}$'\n'${WINE_MONO} == "${geckomono}" ]] ||
+		die "gecko/mono mismatch, has:" ${geckomono}
+
+	default
+
+	# ensure .desktop calls this variant + slot
+	sed -i "/^Exec=/s/wine /${P} /" loader/wine.desktop || die
+
+	# similarly to staging, append to `wine --version` for identification
+	sed -i "s/wine_build[^1]*1/& (Proton-${WINE_PV})/" configure.ac || die
+
+	# source has outdated auto-generated files, update like Proton's Makefile
+	eautoreconf
+	tools/make_requests || die # perl
+	dlls/winevulkan/make_vulkan -x vk.xml || die # python
+}
+
+src_configure() {
+	WINE_PREFIX=/usr/lib/${P}
+	WINE_DATADIR=/usr/share/${P}
+
+	local conf=(
+		--prefix="${EPREFIX}"${WINE_PREFIX}
+		--datadir="${EPREFIX}"${WINE_DATADIR}
+		--includedir="${EPREFIX}"/usr/include/${P}
+		--libdir="${EPREFIX}"${WINE_PREFIX}
+		--mandir="${EPREFIX}"${WINE_DATADIR}/man
+
+		# upstream (Valve) doesn't really support misc configurations (e.g.
+		# adds vulkan code not always guarded by --with-vulkan), so force
+		# some major options that are typically needed by games either way
+		--with-freetype
+		--with-mingw # needed by many, notably Blizzard titles
+		--with-opengl
+		--with-vulkan
+		--with-x
+
+		# ...and disable most options unimportant for games and unused by
+		# Proton rather than expose as volatile USEs with little support
+		--disable-tests # does not build and is disabled in Proton's Makefile
+		--without-capi
+		--without-cups
+		--without-gphoto
+		--without-gssapi
+		--without-krb5
+		--without-ldap
+		--without-netapi
+		--without-opencl
+		--without-pcap
+		--without-sane
+		ac_cv_lib_soname_odbc=
+
+		$(use_enable gecko mshtml)
+		$(use_enable mono mscoree)
+		$(use_with alsa)
+		$(use_with fontconfig)
+		$(use_with gstreamer)
+		$(use_with nls gettext)
+		$(use_with openal)
+		$(use_with osmesa)
+		--without-oss # media-sound/oss is not packaged
+		$(use_with pulseaudio pulse)
+		$(use_with sdl)
+		$(use_with ssl gnutls)
+		$(use_with udev)
+		$(use_with udisks dbus) # dbus is only used for udisks
+		$(use_with unwind)
+		$(use_with usb)
+		$(use_with v4l v4l2)
+		$(use_with vkd3d)
+		$(use_with xcomposite)
+		$(use_with xinerama)
+	)
+
+	tc-ld-force-bfd #867097
+	use custom-cflags || strip-flags # can break in obscure ways, also no lto
+	use crossdev-mingw || PATH=${BROOT}/usr/lib/mingw64-toolchain/bin:${PATH}
+
+	# build using upstream's way (--with-wine64)
+	# order matters: configure+compile 64->32, install 32->64
+	local -i bits
+	for bits in $(usev abi_x86_64 64) $(usev abi_x86_32 32); do
+	(
+		einfo "Configuring ${PN} for ${bits}bits in ${WORKDIR}/build${bits} ..."
+
+		mkdir ../build${bits} || die
+		cd ../build${bits} || die
+
+		# CROSSCC_amd64/x86 are unused by Wine, but recognized here for users
+		if (( bits == 64 )); then
+			: "${CROSSCC:=${CROSSCC_amd64:-x86_64-w64-mingw32-gcc}}"
+			conf+=( --enable-win64 )
+		elif use amd64; then
+			conf+=(
```

Comment:
sorry what exactly is this case? and what do we want to do in it?

Reply:
What? No sure what you mean. You mean why the "use amd64" check?

There's a few different scenario, to call first condition `A`, and use amd64 `B`
`x86, abi_x86_32, -abi_x86_64` = uses neither of these, native toolchain without --enable-win64
`amd64, abi_x86_32, -abi_x86_64` = uses B, but doesn't use --with-wine64 (32bit-only on 64bit, essentially wine's default and can make sense for old stuff)
`amd64, abi_x86_32, abi_x86_64` = uses A and B, passes --with-wine64
`amd64, -abi_x86_32, abi_x86_64` = uses A only (64bit-only build)

Reply:
yeah this is what i was asking thanks